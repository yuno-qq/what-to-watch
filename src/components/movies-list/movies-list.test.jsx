import React from 'react'
import {describe, test, expect} from '@jest/globals'
import renderer from 'react-test-renderer'

import MoviesList from './movies-list.jsx'


describe(`<MoviesList> snapshots:`, () => {
  test(`should render movies ["Красный дракон", "Молчание ягнят", "Ганнибал", "Восхождение Ганнибала", "Черное воскресенье", "Кари Мора"]`, () => {
    const movies = [
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        url: `/`,
        previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        name: `Красный дракон`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        url: `/`,
        previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        name: `Молчание ягнят`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        url: `/`,
        name: `Ганнибал`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
        url: `/`,
        name: `Восхождение Ганнибала`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        url: `/`,
        name: `Черное воскресенье`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
      {
        imageSrc: `https://placehold.co/280x175/EEE/31343C`,
        previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        url: `/`,
        name: `Кари Мора`,
        genre: {
          id: `comedies`,
          name: `Comedies`,
        },
      },
    ]

    const tree = renderer.create(<MoviesList movies={movies} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
