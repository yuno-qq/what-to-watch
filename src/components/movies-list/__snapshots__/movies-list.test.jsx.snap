// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList> snapshots: should render movies ["Красный дракон", "Молчание ягнят", "Ганнибал", "Восхождение Ганнибала", "Черное воскресенье", "Кари Мора"] 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Красный дракон
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Молчание ягнят
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Ганнибал
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Восхождение Ганнибала
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Черное воскресенье
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        loop={true}
        muted={true}
        onCanPlay={[Function]}
        onWaiting={[Function]}
        poster="https://placehold.co/280x175/EEE/31343C"
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/"
      >
        Кари Мора
      </a>
    </h3>
  </article>
</div>
`;
